src/Samples/SubModel.Core/Program.fs

```diff
@@ -108,9 +220,26 @@ module CounterWithClock =
       |> Binding.mapModel (fun m -> m.Clock)
       |> Binding.mapMsg ClockMsg
   ]
+  
+type [<AllowNullLiteral>] public CounterWithClockViewModel(initialModel, dispatch) as this =
+  inherit ViewModelBase<CounterWithClock.Model, CounterWithClock.Msg>(initialModel, dispatch)
+
+  new() = CounterWithClockViewModel(CounterWithClock.init (), ignore)
+
+  member _.Counter = this.subModel((fun m -> m.Counter |> ValueSome), (fun _ -> CounterWithClock.Msg.CounterMsg), CounterViewModel)
+  member _.Clock = this.subModel((fun m -> m.Clock |> ValueSome), (fun _ -> CounterWithClock.Msg.ClockMsg), ClockViewModel)
```

Comment:
This example is important because it illustrates what is necessary for submodel. Basically we need a getter and a message mapper, as well as a view model factory. These view model factories are always supplied with an initial model and a dispatch function.